### ISSUE TITLE
[Tooling] Make It Easier to Specify a 'zap' Instance Outside of the One in 'PATH' for 'run_zaptool.sh'

### ISSUE THREAD

**Issue by gerickson** [project-chip/connectedhomeip#35549]
### Reproduction steps / Feature

In the the Matter documentation at https://project-chip.github.io/connectedhomeip-doc/getting_started/zap.html, it discusses running the `zap` tool via _scripts/tools/zap/run_zaptool.sh_. The `run_zaptool.sh` defaults to wanting to run the `zap` in `PATH`. However, if someone would like to run a `zap` from somewhere else, it is not easily achieved, though one could do:

```
ZAP_INSTALL_PATH=<parent path to zap> scripts/tools/zap/run_zaptool.sh
```

if the user is running a Bourne- or POSIX-compatible shell. However, the above method is not easily discoverable and is not documented outside of the `run_zaptool.sh` code itself.

Add a `--zap` option to allow specifying the name or path to a `zap` executable outside of the `zap` that might be otherwise found in `PATH`.

This relates to issue #35546.

### Platform

other

### Type

Manually tested with SDK

### Anything else?

See above documentation at https://project-chip.github.io/connectedhomeip-doc/getting_started/zap.html.

**Comment by gerickson** [comment]
@bzbarsky-apple, thanks for the insights and the links.

Even for someone reasonably initiated in the technology and building embedded systems, I've found myself tripping over the documentation as I attempt to answer the question: "I am building an embedded, Linux-based Matter device, what are the series of steps (that is, the recipe) I need to undertake to do that?"

Do you anticipate opposition to adding a `--zap-cli` option to _scripts/tools/zap/zap_execution.py_ which appears to execute `zap-cli`. It looks like, were this to move forward, I'd need to also update the documentation at _docs/code_generation.md_ to note `--zap` as another option.

**Comment by gerickson** [comment]
Closed with #35582.
